test/unit/colorspace_spec.js

```diff
@@ -0,0 +1,106 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ColorSpace } from '../../src/core/colorspace';
+
+describe('colorspace', function () {
+  class Test {
+    constructor(decode, n) {
+      this.decode = decode;
+      this.n = n;
+    }
+  }
+  it('should be true if decode is not an array', function() {
+    let test1 = new Test('I am a string', 0);
+    let result1 = true;
+    let returnValue1 = ColorSpace.isDefaultDecode(test1.decode, test1.n);
+
+    let test2 = new Test(42, 0);
+    let result2 = true;
+    let returnValue2 = ColorSpace.isDefaultDecode(test2.decode, test2.n);
+
+    let test3 = new Test(true, 0);
+    let result3 = true;
+    let returnValue3 = ColorSpace.isDefaultDecode(test3.decode, test3.n);
+
+    let test4 = new Test(undefined, 0);
+    let result4 = true;
+    let returnValue4 = ColorSpace.isDefaultDecode(test4.decode, test4.n);
+
+    let test5 = new Test({}, 0);
+    let result5 = true;
+    let returnValue5 = ColorSpace.isDefaultDecode(test5.decode, test5.n);
+
+    expect(returnValue1).toEqual(result1);
+    expect(returnValue2).toEqual(result2);
+    expect(returnValue3).toEqual(result3);
+    expect(returnValue4).toEqual(result4);
+    expect(returnValue5).toEqual(result5);
+  });
+  it('should be true if length of decode array is not correct', function() {
+    let test1 = new Test([0], 1);
+    let result1 = true;
+    let returnValue1 = ColorSpace.isDefaultDecode(test1.decode, test1.n);
+
+    let test2 = new Test([0, 1, 0], 1);
+    let result2 = true;
+    let returnValue2 = ColorSpace.isDefaultDecode(test2.decode, test2.n);
+
+    expect(returnValue1).toEqual(result1);
+    expect(returnValue2).toEqual(result2);
+  });
+  it('should be true if decode map matches the default decode map', function() {
+    let test1 = new Test([], 0);
+    let result1 = true;
+    let returnValue1 = ColorSpace.isDefaultDecode(test1.decode, test1.n);
+
+    let test2 = new Test([0, 0], 1);
+    let result2 = false;
+    let returnValue2 = ColorSpace.isDefaultDecode(test2.decode, test2.n);
+
+    let test3 = new Test([0, 1], 1);
+    let result3 = true;
+    let returnValue3 = ColorSpace.isDefaultDecode(test3.decode, test3.n);
+
+    let test4 = new Test([1, 0], 1);
+    let result4 = false;
+    let returnValue4 = ColorSpace.isDefaultDecode(test4.decode, test4.n);
+
+    let test5 = new Test([1, 1], 1);
+    let result5 = false;
+    let returnValue5 = ColorSpace.isDefaultDecode(test5.decode, test5.n);
+
+    let test6 = new Test([0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 1, 0, 0, 1], 8);
```

Comment:
In practice, I don't think the number of components will usually go beyond `4`, so you can probably use slightly shorter array here (and below too).